src/unsafe-rust/unsafe-extern-c-functions.md

```diff
@@ -0,0 +1,33 @@
+---
+minutes: 3
+---
+
+# Unsafe External Functions
+
+All functions implemented in a foreign language are considered unsafe in Rust:
+
+```rust,editable
+extern "C" {
+    fn abs(input: i32) -> i32;
+}
+
+fn main() {
+    // SAFETY: `abs` doesn't deal with pointers and doesn't have any safety
+    // requirements.
+    unsafe {
+        println!("Absolute value of -3 according to C: {}", abs(-3));
+    }
+}
+```
+
+<details>
+
+`abs` is unsafe because it is an external function (FFI). Calling external
+functions is usually only a problem when those functions do things with pointers
+which might violate Rust's memory model, but in general any C function might
+have undefined behaviour under any arbitrary circumstances.
```

Comment:
Is it worth mentioning here that there is no automatic checking of the Rust function signature against that of the target function? For example, in this case the C function might take `uint64_t` and this would cause UB.